### ISSUE TITLE
Feature query: ban by hostname | mac address | NetBIOS ID

### ISSUE THREAD

**Issue by dsmithhfx** [fail2ban/fail2ban#2354]
Slightly odd but non-critical situation. We are temporarily hosting Windows file and mail servers (two physical servers) on our lan. After considerable difficulty, our Windows tech managed to configure the mail server -- only after calling Microsoft, and discovering there was malware on the server (stay with me: this may/not be relevant to my query).

Every since the Windows servers were plugged into our lan, my wan-facing ubuntu web server has been getting bombarded with hundreds of:
```
192.168.0.1 - - [22/Feb/2019:09:48:07 -0500] "OPTIONS /Microsoft-Server-ActiveSync HTTP/1.1" 200 2192 "-" "Apple-iPhone9C3/1604.57"" 192.168.0.1 is the DHCP gateway.
```
Obviously I can't add a rule to ipban the gateway, so I'm pondering alternatives... is it possible that F2B/IPTABLES supports other ban identifiers (see subject ^^^). I suppose IPTABLES makes the most sense, because I can easily add a rule specific to the known identifier (MAC/NetBIOS/Hostname), if such is permissable.

Sorry if this is too genral or off-topic for F2B!

**Comment by sebres** [comment]
It is not off-topic, but how-to (and you could find it if you'd use search).

Yes it is basically possible (partially in 0.9, and fully implemented since 0.10).

See [How to ban something other as host (IP address), like user or mail, etc. · Wiki](https://github.com/fail2ban/fail2ban/wiki/How-to-ban-something-other-as-host-(IP-address),-like-user-or-mail,-etc.) for details.
This is as regards the fail2ban self - so again, it is possible to recognize the failure for ID whatever you want it would be, and to supply this ID to the service you want use for ban-purposes.

As for the possible ban-action (iptables/etc) you should rather read man-page or documentation or ask on some platforms like serverfault, stackoverflow and superuser.
Basically fail2ban can execute everything on ban/unban, what you can also execute manually from shell.
You could write your own action used your own firewall scripts / scenario's:
```ini
[Definition]
actionban   = my-ban.sh   --ban-mac-addr "<fid>"
actionunban = my-ban.sh --unban-mac-addr "<fid>"
```
More details you'd find in our wiki (see link to article above).

**Comment by sebres** [comment]
And as for the hostname: it works already now (if it can be resolved by your DNS-service).
Token `<HOST>` recognizes IPs but also hostnames, if `usedns` set to `yes` or `warn` (default).
Just find the way to do the ban (proper ban-action accessing firewall/etc, which would prevent such attempts in the future). The tag `<ip>` can be used in action and it'd be substituted to resolved IP from the hostname.

I don't know what exactly may be the hostname in your log-message, so I have simply replaced a part of a name with `localhost`:

```bash
$ fail2ban-regex -v --usedns=yes '192.168.0.1 - - [22/Feb/2019:09:48:07 -0500] "OPTIONS /Microsoft-Server-ActiveSync HTTP/1.1" 200 2192 "-" "localhost/1604.57"" 192.168.0.1 is the DHCP gateway.' '^\S+\s+\S+\s+\S+(?:\s+\[\])?\s+"OPTIONS /Microsoft-Server-ActiveSync [^"]+"\s+\d+\s+\d+\s+\S+\s+"<HOST>/\S+"'

Running tests
=============

Use   failregex line : ^\S+\s+\S+\s+\S+(?:\s+\[\])?\s+"OPTIONS /Microsoft...
Use      single line : 192.168.0.1 - - [22/Feb/2019:09:48:07 -0500] "OPTI...

Results
=======

Failregex: 2 total
|-  #) [# of hits] regular expression
|   1) [2] ^\S+\s+\S+\s+\S+(?:\s+\[\])?\s+"OPTIONS /Microsoft-Server-ActiveSync [^"]+"\s+\d+\s+\d+\s+\S+\s+"<HOST>/\S+"
|      127.0.0.1  Fri Feb 22 15:48:07 2019
|      ::1  Fri Feb 22 15:48:07 2019
`-
...
Lines: 1 lines, 0 ignored, 1 matched, 0 missed
```

As you can see it found `127.0.0.1` and `::1` so fail2ban would execute a ban-action for both this IPs.

If you would test it with `--usedns=raw` instead, you'd see a localhost as name (so fail2ban does not resolve it via DNS), and exactly this (localhost) would be then substituted in your action in `<ip>` or `<fid>` interpolations:
```bash
$ fail2ban-regex -v --usedns=raw ...
...
Failregex: 1 total
|-  #) [# of hits] regular expression
|   1) [1] ^\S+\s+\S+\s+\S+(?:\s+\[\])?\s+"OPTIONS /Microsoft-Server-ActiveSync [^"]+"\s+\d+\s+\d+\s+\S+\s+"<HOST>/\S+"
|      localhost  Fri Feb 22 15:48:07 2019
`-
...
```
